tests/integration/solidity_contracts/UniswapV2/conftest.py

```diff
@@ -0,0 +1,72 @@
+from typing import Callable
+
+import pytest_asyncio
+
+TOTAL_SUPPLY = 10000 * 10**18
+
+
+@pytest_asyncio.fixture(scope="module")
+async def token_a(
+    deploy_solidity_contract: Callable,
+    owner,
+):
+    return await deploy_solidity_contract(
+        "UniswapV2",
+        "ERC20",
+        TOTAL_SUPPLY,
+        caller_address=owner.starknet_address,
+    )
+
+
+@pytest_asyncio.fixture(scope="module")
+async def token_b(
+    deploy_solidity_contract: Callable,
+    owner,
+):
+    return await deploy_solidity_contract(
+        "UniswapV2",
+        "ERC20",
+        TOTAL_SUPPLY,
+        caller_address=owner.starknet_address,
+    )
+
+
+@pytest_asyncio.fixture(scope="module")
+async def factory(
+    deploy_solidity_contract: Callable,
+    owner,
+):
+    return await deploy_solidity_contract(
+        "UniswapV2",
+        "UniswapV2Factory",
+        owner.address,
+        caller_address=owner.starknet_address,
+    )
+
+
+@pytest_asyncio.fixture(scope="module")
+async def pair(
+    deploy_solidity_contract: Callable,
+    token_a,
+    token_b,
+    factory,
+):
+    # TODO: the fixture should use factory.createPair but this currently fails
+    # TODO: with OUT_OF_RESOURCES so we do it "manually" for the sake of running
```

Comment:
do you have a model / explanation for why the `factory.createPair` fails but manual doesn't? i am curious.

Reply:
it does some more ops so if we're close to the limit we then are just above the 1M step. This needs to be confirmed though, I'll push the Univ2Pair tests in a forthcoming PR